src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -5,493 +5,121 @@
-                'AWS.output.starting.sam.app.locally',
-                'Starting the SAM Application locally (see Terminal for output)'
-            )
-        )
-
-        const eventPath: string = path.join(await this.getBaseBuildFolder(), 'event.json')
-        const environmentVariablePath = path.join(await this.getBaseBuildFolder(), 'env-vars.json')
-        const config = await this.getConfig()
-
-        await writeFile(eventPath, JSON.stringify(config.event || {}))
-        await writeFile(
-            environmentVariablePath,
-            JSON.stringify(this.getEnvironmentVariables(config))
-        )
-
-        const command = new SamCliLocalInvokeInvocation(
-            LocalLambdaRunner.TEMPLATE_RESOURCE_NAME,
-            samTemplatePath,
-            eventPath,
-            environmentVariablePath,
-            (!!this._debugPort) ? this._debugPort.toString() : undefined,
-            this.taskInvoker
-        )
-
-        await command.execute()
-
-        if (this.localInvokeArgs.debug) {
-            this.outputChannel.appendLine(
-                localize(
-                    'AWS.output.sam.local.waiting',
-                    'Waiting for SAM Application to start before attaching debugger...'
-                )
-            )
-
-            const timeoutMillis = this.configuration.readSetting<number>(
-                'samcli.debug.attach.timeout.millis',
-                LocalLambdaRunner.SAM_LOCAL_PORT_CHECK_RETRY_TIMEOUT_MILLIS_DEFAULT)
-
-            await tcpPortUsed.waitUntilUsed(
-                this.debugPort,
-                LocalLambdaRunner.SAM_LOCAL_PORT_CHECK_RETRY_INTERVAL_MILLIS,
-                timeoutMillis
-            )
-
-            await this.attachDebugger(this.debugPort)
-        }
-    }
-
-    private async getConfig(): Promise<HandlerConfig> {
-        const workspaceFolder = vscode.workspace.getWorkspaceFolder(this.localInvokeArgs.document.uri)
-        if (!workspaceFolder) {
-            return buildHandlerConfig()
-        }
-
-        const config: HandlerConfig = await getLocalLambdaConfiguration(
-            workspaceFolder,
-            this.localInvokeArgs.handlerName
-        )
-
-        return config
-    }
+    return path.dirname(packageJsonPath)
+}
 
-    private getEnvironmentVariables(config: HandlerConfig): SAMTemplateEnvironmentVariables {
-        if (!!config.environmentVariables) {
-            return {
-                [LocalLambdaRunner.TEMPLATE_RESOURCE_NAME]: config.environmentVariables
+export const initialize = ({
+    configuration,
+    toolkitOutputChannel,
+    processInvoker = new DefaultSamCliProcessInvoker(),
+    taskInvoker = new DefaultSamCliTaskInvoker()
+}: CodeLensProviderParams): void => {
+    vscode.commands.registerCommand(
+        getInvokeCmdKey('javascript'),
+        async (args: LambdaLocalInvokeArguments) => {
+            let debugPort: number | undefined
+
+            if (args.debug) {
+                debugPort = 5858
```

Comment:
Please retain the comment from typescriptCodeLensProvider.ts:
`// TODO : in the future, move this to a utility class and search for an available port`

Reply:
✔